src/lib/ast.rs

```diff
@@ -502,6 +502,17 @@ impl<U: PartialEq + Copy> NodeId<U> {
             .unwrap() + 1
     }
 
+    /// Get the size of the subtree under this node.
+    ///
+    /// The size of a leaf node is 1, the size of a branch node is 1 + the
+    /// number of descendants of that node.
+    pub fn size<T: Clone>(&self, arena: &Arena<T, U>) -> u32 {
+        if self.is_leaf(arena) {
+            return 1;
+        }
+        1 + self.descendants(arena).collect::<Vec<NodeId<U>>>().len() as u32
```

Comment:
Can we use `.count()` rather than creating a vector (which is going to be very slow)?